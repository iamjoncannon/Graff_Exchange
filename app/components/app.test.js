import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import App from './App'
import configureStore from 'redux-mock-store';
import thunk from "redux-thunk"
const mockStore = configureStore([thunk]);
import { Router } from "react-router-dom";
import { createMemoryHistory } from 'history'

const store_setup = (store, pathname) => {

    const history = createMemoryHistory()

    history.push(pathname)

    return renderer.create(
        <Provider store={store} >
           <Router history={history}>
               <App />
           </Router>
       </Provider>
       )
}

describe("app", ()=>{

    it('1) renders successfully with data from redux store', ()=>{
        
        const store = mockStore( {User_state : {isLoggedIn : false}} );
    
        const component = store_setup(store, "/")

        let tree = component.toJSON();
        
        expect(tree).toMatchSnapshot();
    })

    it('if props.location.pathname === "/", renders landing and footer components inside "app-container" div', ()=>{

        const store = mockStore( {User_state : {isLoggedIn : false}} );
    
        const component = store_setup(store, "/")

        const landing = component.root.find(node => node.type.name === "Landing")        
        
        const footer = component.root.find(node => node.type.name === "Footer")        
    
        expect(landing).toBeTruthy()
        
        expect(footer).toBeTruthy()
    })

    it('if not logged in, any path other than "/" redirects to "/"', ()=>{
        
        const store = mockStore( { User_state : {isLoggedIn : false} } );
    
        const component = store_setup(store, "/yada")

        const _router = component.root.find(node => node.type.name === "Router")
        
        expect(_router.props.history.location.pathname).toEqual("/")
    })

    it('>> if logged in <<', ()=>{})

    it('if path includes "/indiv", renders MainNav and IndivNav components', (done)=>{
        
        const store = mockStore( { User_state : { isLoggedIn : true },
                                   Portfolio_state :  { portfolio: {}}
                                 } 
                                );

        const component = store_setup(store, "/indiv")

        const MainNav = component.root.find(node => node.type.name === "MainNav")        
        
        expect(MainNav).toBeTruthy()
        
        const IndivNav = component.root.find(node => node.type.name === "IndivNav")        
    
        expect(IndivNav).toBeTruthy()

        done()
    })
    
    it('if path includes "/all", renders MainNav and AllNav components', (done)=>{

        const store = mockStore( { User_state : { isLoggedIn : true },
            Portfolio_state :  { portfolio: {}}
          } 
         );

        const component = store_setup(store, "/all")

        const MainNav = component.root.find(node => node.type.name === "MainNav")        

        expect(MainNav).toBeTruthy()

        const AllNav = component.root.find(node => node.type.name === "AllNav")        

        expect(AllNav).toBeTruthy()

        done()
    })
})