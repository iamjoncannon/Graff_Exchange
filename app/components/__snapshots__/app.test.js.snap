// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app 1) renders successfully with data from redux store 1`] = `
<div
  className="app-container"
>
  <div
    className="landing"
  >
    <div
      className="header"
    >
      <span
        className="title"
      >
        Graff Exchange
      </span>
      <span
        className="blurb"
      >
        Graff Exchange, a real-time stock portfolio application built with a full stack: React, Redux, Sass, Socket.io, Node.js, GraphQL, Redis, PostgreSQL, and AWS.
      </span>
      <img
        src="../assets/graphql_logo.png"
      />
      <div
        className="sign-btns"
      >
        <span
          onClick={[Function]}
        >
          Sign In
        </span>
        <span
          className="unselected"
          onClick={[Function]}
        >
          Sign Up
        </span>
      </div>
    </div>
    <div
      className="form"
    >
      <form>
        <div>
          <span>
            Email
          </span>
          <input
            name="email"
            onChange={[Function]}
            required={true}
            type="email"
            value=""
          />
        </div>
        <div>
          <span>
            Password
          </span>
          <input
            name="password"
            onChange={[Function]}
            type="password"
            value=""
          />
        </div>
        <button
          onClick={[Function]}
        >
          Sign In
        </button>
        <span
          onClick={[Function]}
        >
          Sign into Demo Account
        </span>
      </form>
    </div>
  </div>
  <footer>
    <a
      href="https://github.com/iamjoncannon/Graff_Exchange"
      target="_blank"
    >
      <img
        src="https://jayceecodesphotos.s3.amazonaws.com/react.png"
      />
    </a>
    <a
      href="https://github.com/iamjoncannon/Graff_Exchange"
      target="_blank"
    >
      <img
        src="https://jayceecodesphotos.s3.amazonaws.com/redux.png"
      />
    </a>
    <a
      href="https://github.com/iamjoncannon/Graff_Exchange"
      target="_blank"
    >
      <img
        src="https://upload.wikimedia.org/wikipedia/commons/thumb/9/96/Sass_Logo_Color.svg/2000px-Sass_Logo_Color.svg.png"
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": "10px",
          }
        }
      />
    </a>
    <a
      href="https://github.com/iamjoncannon/Graff_Exchange_API"
      target="_blank"
    >
      <img
        src="../assets/graphql_logo.png"
      />
    </a>
    <a
      href="https://github.com/iamjoncannon/Graff_Exchange_API"
      target="_blank"
    >
      <img
        src="http://pluspng.com/img-png/nodejs-logo-png-nice-images-collection-node-js-desktop-wallpapers-370.png"
      />
    </a>
    <a
      href="https://github.com/iamjoncannon/Graff_Exchange_API"
      target="_blank"
    >
      <img
        src="https://cdn.iconscout.com/icon/free/png-512/redis-6-1175105.png"
      />
    </a>
    <a
      href="https://github.com/iamjoncannon/Graff_Exchange_API"
      target="_blank"
    >
      <img
        src="https://jayceecodesphotos.s3.amazonaws.com/postgres.png"
      />
    </a>
    <a
      href="https://github.com/iamjoncannon/Graff_Exchange_API"
      target="_blank"
    >
      <img
        src="https://jayceecodesphotos.s3.amazonaws.com/aws.png"
        style={
          Object {
            "borderRadius": "10px",
          }
        }
      />
    </a>
  </footer>
</div>
`;
